types/node/index.d.ts

```diff
@@ -780,101 +787,60 @@ declare module "http" {
         socketPath?: string;
         method?: string;
         path?: string;
-        headers?: { [key: string]: any };
+        headers?: OutgoingHeaders;
         auth?: string;
         agent?: Agent | boolean;
         timeout?: number;
     }
 
-    export interface Server extends net.Server {
+    export class Server extends net.Server {
+        constructor(req: IncomingMessage, res: ServerResponse);
         setTimeout(msecs: number, callback: Function): void;
         maxHeadersCount: number;
         timeout: number;
         listening: boolean;
     }
-    /**
-     * @deprecated Use IncomingMessage
-     */
-    export interface ServerRequest extends IncomingMessage {
-        connection: net.Socket;
-    }
-
-    export interface ServerResponseHeaders {
-      [key: string]: number | string | string[];
-    }
-
-    export interface ServerResponse extends stream.Writable {
-        // Extended base methods
-        write(buffer: Buffer): boolean;
-        write(buffer: Buffer, cb?: Function): boolean;
-        write(str: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, cb?: Function): boolean;
-        write(str: string, encoding?: string, fd?: string): boolean;
 
-        addTrailers(headers: ServerResponseHeaders): void;
+    export class OutgoingMessage extends stream.Writable {
         finished: boolean;
-        getHeader(name: string): number | string | string[] | undefined;
-        getHeaderNames(): string[]
-        getHeaders(): any
-        hasHeader(name: string): boolean
+        sendDate: boolean;
         headersSent: boolean;
+
+        constructor();
+        setTimeout(msecs: number, callback?: () => void): this;
+        setHeader(name: string, value: number | string | string[]): void;
+        getHeader(name: string): number | string | string[] | undefined;
         removeHeader(name: string): void;
-        sendDate: boolean;
-        setHeader(name: string, value: string | string[]): void;
-        setTimeout(msecs: number, callback: Function): ServerResponse;
+        addTrailers(headers: OutgoingHeaders): void;
+    }
```

Comment:
Streams have additionally support for object mode which allows any object. HTTP allows only `string' and `Buffer` (and maybe `Uint8Array`).